### PR TITLE
Config: Add support for SOLID linting and remove Prettier

### PR DESCRIPTION
**Pull request by poteat**
## What was changed?

Prettier was removed in favor of direct ESLint formatting rules of a similar scope & behavior. As well, support for SOLID-related linting rules surrounding decorators were added, which are currently in use in the updated QOTD codebase.

I also took the opportunity to improve the documentation and remove legacy concerns.

The majority of code which previously passed or failed should still do so.

## Why was this change needed?

Configuring direct ESlint formatting rules allows a higher amount of control and customization as opposed to Prettier.

## Notes

-   [X] Was this tested?
      - Yes, I ran this successfully on the `solid-monorepo/solid-scenes` codebase
-   [ ] Can a unit/integration test be written?
      - Maybe, I've been thinking about the possibility of adding some example code in this codebase which the linting configuration runs against, in order to detect changes in linting behavior.
-   [X] Is documentation updated to reflect this change?
      - Yes

### CODE REVIEW

File: templates/.eslintrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (angeldehart) [inline templates/.eslintrc:1]: do we need this file anymore? copying from templates isn't mentioned in the readme anymore.

**Review by angeldehart — APPROVED** [review]
🔋

**Comment by willthefirst** [comment]
We are getting a build error from this update: https://github.com/Volley-Inc/qotd-solid-gev/actions/runs/3840530382/jobs/6539698789

I think we need those packages in `dependencies`? Unless you wanted to make us add them explicitly as dev dependencies in the consumer repo?

**Comment by poteat** [comment]
Ah yeah you'll need to install additional dependencies in your package. ESLint configurations do not "transitively transfer" dependencies, so you need to replicate them on your side. @willthefirst

**Comment by willthefirst** [comment]
> Ah yeah you'll need to install additional dependencies in your package. ESLint configurations do not "transitively transfer" dependencies, so you need to replicate them on your side. @willthefirst

Why not at least make them peer dependencies then? I even think this was working with those packages listed as normal dependencies (but I could be wrong).

I ask because as it stands, the README doesn't tell you much about how to install this package. I'm all for simplicity and minimizing required setup, but right now the instructions leave the consumer with build errors. I'd bring back at least some of the README and use dependencies as much as possible.

**Pull request merged**